src/main/java/com/avispl/symphony/dal/communicator/lg/lcd/LgLCDDevice.java

```diff
@@ -974,10 +1044,11 @@ private void populateMonitoringAndControllingData() throws InterruptedException
 						timeoutCount++;
 					}
 					//If the Future is not completed after the defaultConfigTimeout =>  update the failedMonitor and destroy the connection.
-					if (!devicesExecutionPool.get(finalIndexFuture).isDone()) {
+					int len = devicesExecutionPool.size() - 1;
```

Comment:
is the variable "len" represent length? but in this case, it should be the lastIndex.